mmdet/core/bbox/assigners/atss_cost_assigner.py

```diff
@@ -0,0 +1,199 @@
+    - positive integer: positive sample, index (1-based) of assigned gt
+
+    Args:
+        topk (int): number of bbox selected in each level.
+        alpha (float): param of cost rate for each proposal. Default 0.8.
+        iou_calculator (dict): builder of IoU calculator.
+            Default dict(type='BboxOverlaps2D').
+        ignore_iof_thr (int): whether ignore max overlaps or not.
+            Default -1 (1 or -1).
+    """
+
+    def __init__(self,
+                 topk,
+                 alpha=0.8,
+                 iou_calculator=dict(type='BboxOverlaps2D'),
+                 ignore_iof_thr=-1):
+        self.topk = topk
+        self.alpha = alpha
+        self.iou_calculator = build_iou_calculator(iou_calculator)
+        self.ignore_iof_thr = ignore_iof_thr
+
+    # https://github.com/sfzhang15/ATSS/blob/master/atss_core/modeling/rpn/atss/loss.py
+    @torch.no_grad()
+    def assign(self,
+               bboxes,
+               num_level_bboxes,
+               cls_scores,
+               bbox_preds,
+               gt_bboxes,
+               gt_bboxes_ignore=None,
+               gt_labels=None):
+        """Assign gt to bboxes.
+
+        The assignment is done in following steps
+
+        1. compute iou between all bbox (bbox of all pyramid levels) and gt
+        2. compute center distance between all bbox and gt
+        3. on each pyramid level, for each gt, select k bbox whose center
+           are closest to the gt center, so we total select k*l bbox as
+           candidates for each gt
+        4. get corresponding iou for the these candidates, and compute the
+           mean and std, set mean + std as the iou threshold
+        5. select these candidates whose iou are greater than or equal to
+           the threshold as positive
+        6. limit the positive sample's center in gt
+
+        Args:
+            bboxes (Tensor): Bounding boxes to be assigned, shape(n, 4).
+            num_level_bboxes (List): num of bboxes in each level
+            cls_scores (Tensor): Class scores, shape(n, 4).
+            bbox_preds (Tensor): Bounding boxes which are predicted,
+                shape(n, 4).
+            gt_bboxes (Tensor): Groundtruth boxes, shape (k, 4).
+            gt_bboxes_ignore (Tensor, optional): Ground truth bboxes that are
+                labelled as `ignored`, e.g., crowd boxes in COCO.
+            gt_labels (Tensor, optional): Label of gt_bboxes, shape (k, ).
+
+        Returns:
+            :obj:`AssignResult`: The assign result.
+        """
+        INF = 100000000
+        # NOTE first convert anchor to prediction bbox
+        bboxes = bboxes[:, :4]  # anchor bbox
+
+        num_gt, num_bboxes = gt_bboxes.size(0), bboxes.size(0)
+
+        # NOTE DeFCN style cost function
+        # compute iou between all bbox and gt
+        overlaps = self.iou_calculator(bbox_preds, gt_bboxes)
+        # compute cls cost for bbox and GT
+        cls_cost = torch.sigmoid(cls_scores[:, gt_labels])
```

Comment:
Variable names need to conform to mathematical conventions. Usually, the input of the sigmoid should be called "logits" and the output should be called "probability".

Reply:
```suggestion
        cls_score = torch.sigmoid(cls_logits[:, gt_labels])
```

or

```suggestion
        cls_score = torch.sigmoid(cls_pred[:, gt_labels])
```